### PR TITLE
webpack: Loader API

### PR DESCRIPTION
**Pull request by shlomiassaf**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://webpack.js.org/api/loaders/
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: webpack/index.d.ts [hunk 1]
```diff
@@ -14,11 +14,11 @@ declare namespace webpack {
         context?: string;
         entry?: string | string[] | Entry;
         /** Choose a developer tool to enhance debugging. */
-        devtool?: string;
+        devtool?: 'cheap-eval-source-map' | 'cheap-source-map' | 'cheap-module-eval-source-map' | 'cheap-module-source-map'| 'eval-source-map'| 'source-map' | 'nosources-source-map';
```
Comment (mohsen1) [inline webpack/index.d.ts]: ❤️

**Comment by shlomiassaf** [comment]
Fixing the issues, I didn't know about the tests...

**Comment by shlomiassaf** [comment]
LOL, can't run the tests, `nodegit` won't install on my mac... i'll see what I can do with it :)

**Comment by shlomiassaf** [comment]
OK, my forked `master` was way to old, lots of changes with upper/lower case file names that OSX and git can't handle together.

I'll wipe out my fork and repost the pull request.

**Pull request closed without merging**